src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts

```diff
@@ -401,6 +404,49 @@ export class NfsFormComponent extends CdForm implements OnInit {
     this.nfsForm.patchValue(res);
     this.setPathValidation();
     this.clients = res.clients;
+
+    if (this.isEdit) {
+      this.setUpVolumeValidation();
+
+      if (res.fsal.name === SUPPORTED_FSAL.CEPH && res.path) {
+        this.resolveCephfsPath(res.path, res.fsal.fs_name);
+      } else {
+        this.resolveCephfsPath('', res.fsal.fs_name);
+      }
+    }
+  }
+
+
+  // Setting up Subvolumegroup and Subvolume for CephFS paths on Edit.
+  resolveCephfsPath(path: string, fsName: string) {
+    this.getSubVolGrp(fsName);
+    if (!path?.startsWith('/volumes/')) return;
+
+    const splitPath = path.split('/');
+    const subvolume_group = splitPath[2];
+    const subvolume = splitPath[3];
+
+    this.getSubVolGrp(fsName);
+
+    const subvolGroup = subvolume_group === this.defaultSubVolGroup ? '' : subvolume_group;
+    this.subvolService.get(fsName, subvolGroup).subscribe((subvols: any[]) => {
+      this.allsubvols = subvols;
+
+      const subvolume1 = subvols.find((s) => s.name === subvolume);
+      if (subvolume1) {
+        this.nfsForm.patchValue({ subvolume });
+      }
```

Comment:
We should set it to initial state, if the subvolume is not found.

Then this might not be required.

```
    if (this.isEdit) {
      this.nfsForm.get('subvolume').setValue('');
    }
```